pokemon/src/main/java/gamarket/EncounterGUI.java

```diff
@@ -1,125 +1,1075 @@
+    private Button move1, move2, move3, move4;
+    private boolean done, runStatus;
+    private Button hp1, hp2; 
+    private Button fight, bag, pokemon, run, box; 
+    private GridPane gp, poke, menu, textlols, playerSide, wildSide; 
+    private VBox battleMenu, fightMenu;  
+    private Text text;
+    static Pokemon wildPokemon, playerPokemon;
+    private String wildPokePath, playerPokePath, attackerStatus, input;
+    private static Encounter aEncounter;
+    static boolean wasMove1Used, wasMove2Used, wasMove3Used, wasMove4Used;
+    private ImageView enemyPoke, playerPoke;
+    private VBox pokemonMenu;
+    private Button pokemons[];
+    private String path, filePath;
+    protected static EncounterGUI encounterGui;
+    private SceneController sceneController;
+
+    public EncounterGUI(Pokemon wildPokemon, Pokemon playerPokemon, Encounter aEncounter) {
+        this.wildPokemon = wildPokemon;
+        this.playerPokemon = playerPokemon;
+        this.aEncounter = aEncounter;
+        pokemons = new Button[6];
+    }
+    
+    public static synchronized EncounterGUI getInstance() {
+        if (encounterGui == null) {
+            encounterGui = new EncounterGUI(wildPokemon, playerPokemon, aEncounter);
+        }
+        return encounterGui;
+    }
+
+    public void setSceneController(Stage window){ this.sceneController = SceneController.getInstance(window);
+        System.out.println("scene controller"+ sceneController);}
+ 
+    public GridPane display() {
+        
+        gp = new GridPane();
+        poke = new GridPane();
+        Soundtrack.stopMusic();
+        Soundtrack.loadMusic("wild_encounter.wav");
+        Soundtrack.startMusic();
+        fightMenu = new VBox(18);
+ 
+        // Set up wild pokemon front image. 
+        this.getFrontPokeImage();
+        // Display player's pokemon, temporary set-up, will use a function to set playerPokePath. 
+        playerPokePath = this.whatPokeBackImage(playerPokemon.getIdentStats().getName());
+        // Background image for the pokemon wild encounter. 
+        filePath = "./pokemon/imgs/battle_background.png";
+        File file = new File(filePath);
+        String path3 = file.toURI().toString();
+        Image image3 = new Image(path3);
+        ImageView background = new ImageView(image3);
+        background.setFitHeight(300);
+        background.setFitWidth(300);
+        BackgroundSize bSize = new BackgroundSize(BackgroundSize.AUTO, 
+                                                    BackgroundSize.AUTO, false, false, true, true);
+        gp.setBackground(new Background(new BackgroundImage(image3, BackgroundRepeat.NO_REPEAT,
+                           BackgroundRepeat.NO_REPEAT, BackgroundPosition.CENTER, bSize)));
+ 
+        // Image for the fight button. 
+        filePath = "./pokemon/imgs/fight_button_image.png";
+        File test = new File(filePath);
+        String x = test.toURI().toString();
+        ImageView a = new ImageView(x);

```

Comment:
this code block gets used 4 times. make into a function that takes a string filepath and returns an imageview